flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -87,6 +88,12 @@
 	/** The number of available buffers that have not been announced to the producer yet. */
 	private final AtomicInteger unannouncedCredit = new AtomicInteger(0);
 
+	/** The number of unsent buffers in the producer's sub partition. */
+	private final AtomicInteger senderBacklog = new AtomicInteger(0);
+
+	/** The tag indicates whether this channel is waiting for additional floating buffers from the buffer pool. */
+	private final AtomicBoolean isWaitingFloatingBuffers = new AtomicBoolean(false);
```

Comment:
better: `isWaitingForFloatingBuffers`

Reply:
but in general, do we need this field? For now, I only see this being used to verify correct use and I'm not sure whether the costs verify this. Also, nothing in this PR sets it to `true`.

Reply:
These two fields are currently used in `notifyBufferAvailable()` logic, so we have to define them in this PR.
In next PR #4509 , `isWaitingForFloatingBuffers` field will be set  true by two conditions:
1. The number of current available buffers is less than sender backlog
2. There are no available floating buffers in `BufferProvider`
And this field also used to avoid register listener in `BufferProvider` multi times.

It may confuse you to only see this PR change, sorry for that.

Reply:
ok, fine - so I'll see more details in the next PR and we get this in here